Assets/Editor/ModelInspector.cs

```diff
@@ -510,30 +684,41 @@ public override void OnPreviewGUI(Rect r, GUIStyle background)
                         "Mesh preview requires\nrender texture support");
                 return;
             }
-
+            
             Init();
 
             Assembly editorAssembly = Assembly.GetAssembly(typeof(EditorGUI));
             Type guiPreview = editorAssembly.GetType("PreviewGUI");
             MethodInfo drag2D = guiPreview.GetMethod("Drag2D");
-
-            if(m_Settings.displayMode != DisplayMode.UVLayout)
-                m_Settings.previewDir = (Vector2)drag2D?.Invoke(null, new object[] {m_Settings.previewDir, r});
-            //previewDir = PreviewGUI.Drag2D(previewDir, r);
             
-            if (Event.current.type == EventType.ScrollWheel && m_Settings.displayMode == DisplayMode.UVLayout)
+            if (Event.current.commandName == "FrameSelected") // compare against EventCommandNames.FrameSelected when on trunk
+            {
+                FrameObject();
+                
+                if(Event.current.type != EventType.Repaint && Event.current.type != EventType.Layout)
```

Comment:
`if ((evt.type == EventType.ValidateCommand || evt.type == EventType.ExecuteCommand) && evt.commandName == "FrameSelected")` in the outer if would be more proper, and then there's no need to do this check here.